### PR TITLE
NP-12 Make OpenMRS Name Phonetic Module support PostgreSQL

### PR DESCRIPTION
**Pull request by Am-Coder**
**Changes I made**
1. Corrected sequence mapping.
2. Removed sqldiff and added equivalent liquibase file.

**Link to ticket**
https://issues.openmrs.org/browse/NP-12

### CODE REVIEW

**Comment by Am-Coder** [comment]
@dkayiwa , I have tested the changes on MySQL as well as PostgreSQL and eveything is working fine. Also, I tried the local build and all the tests are passing.

File: api/src/main/resources/liquibase.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            <column name="field" type="INT">
+                <constraints nullable="false"/>
+            </column>
+            <column name="renderer_class_name" type="VARCHAR(255)">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+    </changeSet>
+
+    <changeSet author="aman (generated)" id="name_phonetics-1598359669281-519">
```
Comment (dkayiwa) [inline api/src/main/resources/liquibase.xml:29]: Did we have this in the original sqldiff?

Comment (Am-Coder) [reply api/src/main/resources/liquibase.xml:29]: MySQL automatically creates an index for a foreign key. Thus this index got created in db and liquibase generated a change set. I found this here - https://dev.mysql.com/doc/refman/8.0/en/constraint-foreign-key.html

Comment (dkayiwa) [reply api/src/main/resources/liquibase.xml:29]: Wouldn't this index also get automatically created by mysql after running the liquibase changeset which creates the foreign key?

Comment (Am-Coder) [reply api/src/main/resources/liquibase.xml:29]: @dkayiwa , you are right. Its obvious. My bad. I was looking at whether PostgreSQL creates indexes for foreign keys and it appears it does not( See [here](https://stackoverflow.com/questions/970562/postgres-and-indexes-on-foreign-keys-and-primary-keys) ). So, I went looking for articles how much search time will be reduced if we set the index. I found this [article](https://www.cybertec-postgresql.com/en/index-your-foreign-key/) and it appears that on large databases we should have this index. If we are having it in MySQL then we should probably have it in PostgreSQL as well.

Its your call whether we should have or remove this index ?

Comment (dkayiwa) [reply api/src/main/resources/liquibase.xml:29]: Fair enough! 😊

Comment (dkayiwa) [reply api/src/main/resources/liquibase.xml:29]: So the next question, why are you using a name that is already used by the foreign key constraint?

Comment (Am-Coder) [reply api/src/main/resources/liquibase.xml:29]: Actually, its not me doing it , its what liquibase generated :sweat_smile: . Similar situations are present all over the schema file. For eg consider constraint name : address_for_person [here](https://github.com/openmrs/openmrs-core/blob/288f9251ecce8997bce4ee92608a23c741f8e135/api/src/main/resources/org/openmrs/liquibase/snapshots/schema-only/liquibase-schema-only-2.3.x.xml#L2975) and [here](https://github.com/openmrs/openmrs-core/blob/288f9251ecce8997bce4ee92608a23c741f8e135/api/src/main/resources/org/openmrs/liquibase/snapshots/schema-only/liquibase-schema-only-2.3.x.xml#L4936)

Comment (dkayiwa) [reply api/src/main/resources/liquibase.xml:29]: Oh i see. 😊

**Pull request merged**